include/swift/AST/DiagnosticsSema.def

```diff
@@ -3293,6 +3293,13 @@ NOTE(availability_conformance_introduced_here, none,
      "conformance introduced here", ())
 
 //------------------------------------------------------------------------------
+// @discardableresult
+//------------------------------------------------------------------------------
+
+WARNING(discardable_result_on_void_function, none,
+"Functions returning void cannot be declared @discardableresult", ())
```

Comment:
The spelling, here as elsewhere, should be `@discardableResult` (also, `Void`). Note also indentation and capitalization of this line (align quotation mark, lowercase f). I also wonder if the right message is "cannot be"--perhaps just "@discardableResult redundant for function returning Void"?